include/dlaf/matrix/copy_tile.h

```diff
@@ -89,5 +122,63 @@ void copy(TileElementSize region, TileElementIndex in_idx, const matrix::Tile<co
   dlaf::tile::lacpy<T>(region, in_idx, in, out_idx, out);
 }
 
+#define DLAF_MAKE_FUNCTION_OBJECT(fname)     \
+  constexpr struct fname##_t {               \
+    template <typename... Ts>                \
+    decltype(auto) operator()(Ts&&... ts) {  \
+      return fname(std::forward<Ts>(ts)...); \
+    }                                        \
+  } fname##_o {                              \
+  }
+
+DLAF_MAKE_FUNCTION_OBJECT(copy);
+
+namespace internal {
+template <typename T>
+struct UnwrapFuture {
+  template <typename U>
+  static decltype(auto) call(U&& u) {
+    return std::forward<U>(u);
+  }
+};
+
+template <typename T, Device D>
+struct UnwrapFuture<hpx::future<Tile<T, D>>> {
+  template <typename U>
+  static auto call(U&& u) {
+    auto t = u.get();
+    return t;
+  }
+};
+
+template <typename T>
+struct dummy;
```

Comment:
What is this for?

Reply:
Debugging... I'll remove it. It prints the type `T` with an error since there's no definition.

Reply:
good to know for template debugging!